b2/arg_parser.py

```diff
@@ -78,9 +83,19 @@ def description(self, value):
 
     def error(self, message):
         self.print_help()
+
         args = {'prog': self.prog, 'message': message}
         self.exit(2, '\n%(prog)s: error: %(message)s\n' % args)
 
+    @classmethod
+    def _get_encoding(cls):
+        _, locale_encoding = locale.getdefaultlocale()
+        if locale_encoding is not None:
+            return sys.getdefaultencoding()
+
+        # locales are improperly configured
```

Comment:
perhaps we should emit a warning in this case? If not, lets at least log it so that if someone runs in `--verbose` mode, they can clearly see what's going on

Reply:
No, it is only for printing help messages or argparse errors. All the parameters, including `--verbose` are not even evaluated in such a case, and the logger is disabled by deault so not one will see it. I could emit a python warning, but I don't think it is a valid case